htdocs/loan/payment/payment.php

```diff
@@ -293,14 +293,14 @@
 		print '</tr>';
 
 		print '<tr><td class="fieldrequired">'.$langs->trans("PaymentMode").'</td><td colspan="2">';
-		$form->select_types_paiements(GETPOSTISSET("paymenttype") ? GETPOST("paymenttype", 'alphanohtml') : $loan->fk_typepayment, "paymenttype");
+		$form->select_types_paiements(GETPOSTISSET("paymenttype") ? GETPOST("paymenttype", 'alphanohtml') : (!empty($line) ? $line->fk_typepayment : ''), "paymenttype");
```

Comment:
Why not just using ' '   ?
$line just does not exists or has no sens on this line. What is $line on a field that is at loan level ?
It seems we dont have fk_typepayment store on loan, so may be we should completely remove this field off the form creation ?

Reply:
When calling payment.php, il will seek the last unpaid scheduled term and fill $line with the related values (lines 54 to 84). So in practice, $line will be defined most of the time you want to enter a payment. Payment type could have been defined as the loan level, but it is not unreasonable to have it at the schedule level either. In any case it is supposed to be a mandatory information when registering payments.

Reply:
So you must use a new var $lastpaymenttypeoflines initialized before the loop
and set it into the loop, so when you are outside the loop this var still has a meaning.
Then use it into select_types_paiements
On line 296 $line is potentially a var that should does not exists anymore since it was defined inside the loop (so var is local to the scope of the loop only). Having it still defined outside of a loop is a side effect but code must not rely on side effect.

Reply:
I am not sure I get your point here. Should'nt the empty test return true if it is not defined ? By the way, the same test is performed 4 other times in that script.